runners/spark/pom.xml

```diff
@@ -134,6 +152,28 @@
             <forkCount>1</forkCount>
             <reuseForks>false</reuseForks>
           </configuration>
+          <executions>
+            <execution>
+              <id>runnable-on-service-tests</id>
+                 <configuration>
+                   <groups>org.apache.beam.sdk.testing.RunnableOnService</groups>
+                <parallel>none</parallel>
+                <failIfNoTests>true</failIfNoTests>
+                <dependenciesToScan>
+                  <dependency>org.apache.beam:java-sdk-all</dependency>
+                </dependenciesToScan>
+                <systemPropertyVariables>
+                  <beamTestPipelineOptions>
+                    [
+                      "--runner=org.apache.beam.runners.spark.TestSparkPipelineRunner",
+                      "--streaming=false"
+                    ]
+                  </beamTestPipelineOptions>
+                  <dataflow.spark.test.reuseSparkContext>true</dataflow.spark.test.reuseSparkContext>
```

Comment:
So I took a deeper look into this "seconds-scale" overhead time we're hitting, and I don't see an easy way avoiding this, though there are some optimizations that we can do (i.e., turn off Spark WebUI), I couldn't find anything that will make a real difference - and that makes sense.
Those "unit-tests" are IT tests actually (though I've heard arguments about this terminology..) that spin-up a Spark instance, which takes a few seconds and that's reasonable, it's the same for Flink (last time I've checked).
This leads to a bigger question - do we want to execute ALL RunnableOnService tests on ALL runners ? because each test that executes a pipeline will probably present some overhead that is more than the Direct/InProcessPipelineRunner will..
I guess it's a trade-off between what could become a very long build, and a less comprehensive testing infrastructure.
We could also consider "grouping" together tests so one pipeline execution tests more than one component, and this is a compromise as well.

Thoughts ?